cmd/operator/rollout.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	runapi "github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/run"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/config"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/rollout"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// runRollouts concurrently handles the rollout of the targeted services.
+func runRollouts(ctx context.Context, logger *logrus.Logger, svcs []*rollout.ServiceRecord, strategy *config.Strategy) []error {
+	var (
+		errs []error
+		mu   sync.Mutex
+		wg   sync.WaitGroup
+	)
+	for _, svc := range svcs {
+		wg.Add(1)
+		go func(ctx context.Context, lg *logrus.Logger, svc *rollout.ServiceRecord, strategy *config.Strategy) {
+			err := handleRollout(ctx, lg, svc, strategy)
+			if err != nil {
+				mu.Lock()
+				errs = append(errs, err)
+				mu.Unlock()
+			}
+			wg.Done()
```

Comment:
let's move to the beginning of the function + defer

Reply:
I agree with this, but I just realized that if we do this, `runDaemon` [can't ever return an error](https://github.com/GoogleCloudPlatform/cloud-run-release-operator/blob/c6b0627606cca06a83a0828fecce8b5042f3cadf/cmd/operator/main.go#L156). 

But this is not a big problem since getting services mostly fail either because 
1. a Cloud Run region is down (like last week) 
2. wrong region is specified (e.g. `us-east`, notice typo)

For 1, we can make the argument that's not the tool's problem and keeping it running is actually the right decision. For 2, the user should quickly realized of the problem since it'd happen for the first time right at the beginning of the execution.